Sources/Kitura/CodableRouter+TypeSafeMiddleware.swift

```diff
@@ -0,0 +1,1245 @@
+    ) {
+        get(route) { request, response, next in
+            Log.verbose("Received GET (singular) type-safe request with middleware and Query Parameters")
+            Log.verbose("Query Parameters: \(request.queryParameters)")
+            // Define result handler
+            do {
+                let query: Q = try QueryDecoder(dictionary: request.queryParameters).decode(Q.self)
+                self.handleMiddleware(T1.self, T2.self, T3.self, request: request, response: response) { typeSafeMiddleware1, typeSafeMiddleware2, typeSafeMiddleware3 in
+                    guard let typeSafeMiddleware1 = typeSafeMiddleware1, let typeSafeMiddleware2 = typeSafeMiddleware2, let typeSafeMiddleware3 = typeSafeMiddleware3 else {
+                        return next()
+                    }
+                    handler(typeSafeMiddleware1, typeSafeMiddleware2, typeSafeMiddleware3, query, CodableHelpers.constructOutResultHandler(response: response, completion: next))
+                }
+            } catch {
+                // Http 400 error
+                response.status(.badRequest)
+                next()
+            }
+        }
+    }
+
+    /**
+     Sets up a closure that will be invoked when a GET request to the provided route is received by the server.
+     The closure accepts a successfully executed instance of `TypeSafeMiddleware`, the parsed query parameters,
+     and a handler which responds with an array of Codable objects or a `RequestError`.
+     The handler contains the developer's logic, which determines the server's response.
+     ### Usage Example: ###
+     In this example, MySession is a struct that conforms to the TypeSafeMiddleware protocol.
+     session is an instance of MySession, that contains an [Int: [User]] dictionary.
+     queryParam is a struct containing the decoded query parameters.
+     User is a struct that conforms to Codable.
+     ```swift
+     struct Query: QueryParams {
+        let id: Int
+     }
+     router.get("/user") { (session: MySession, query: Query, respondWith: ([User]?, RequestError?) -> Void) in
+        guard let user: [User] = session.user[query.id] else {
+            return respondWith(nil, .notFound)
+        }
+        respondWith(user, nil)
+     }
+     ```
+     #### Multiple Middleware: ####
+     The closure can process up to three `TypeSafeMiddleware` objects by defining them in the handler:
+     ```swift
+     router.get("/user") { (middle1: Middle1, middle2: Middle2, middle3: Middle3, query: Query,
+                            respondWith: ([User]?, RequestError?) -> Void) in
+     ```
+     - Parameter route: A String specifying the URL path that will invoke the handler.
+     - Parameter handler: A closure that receives a TypeSafeMiddleware instance and a QueryParams instance, and returns an array of Codable objects or a RequestError.
+     */    public func get<T: TypeSafeMiddleware, Q: QueryParams, O: Codable>(
```

Comment:
This */ is being displayed in the API doc.  `public func...` needs to move onto the next line to fix this.